sdk/identity/Azure.Identity/src/PersistentTokenCache.cs

```diff
@@ -0,0 +1,157 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.Extensions.Msal;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// Persistent token cache.
+    /// </summary>
+    public class PersistentTokenCache : TokenCache
+    {
+        // we are creating the MsalCacheHelper with a random guid based clientId to work around issue https://github.com/AzureAD/microsoft-authentication-extensions-for-dotnet/issues/98
+        // This does not impact the functionality of the cacheHelper as the ClientId is only used to iterate accounts in the cache not for authentication purposes.
+        private static readonly string s_msalCacheClientId = Guid.NewGuid().ToString();
+
+        private static AsyncLockWithValue<MsalCacheHelper> cacheHelperLock = new AsyncLockWithValue<MsalCacheHelper>();
+        private static AsyncLockWithValue<MsalCacheHelper> s_ProtectedCacheHelperLock = new AsyncLockWithValue<MsalCacheHelper>();
+        private static AsyncLockWithValue<MsalCacheHelper> s_FallbackCacheHelperLock = new AsyncLockWithValue<MsalCacheHelper>();
+        private readonly bool _allowUnencryptedStorage;
+        private readonly string _name;
+
+        /// <summary>
+        /// Creation.
+        /// </summary>
+        /// <param name="allowUnencryptedStorage"></param>
+        public PersistentTokenCache(bool allowUnencryptedStorage = true)
```

Comment:
Is this setting that important to have a ctor overload with it?

Reply:
I agree we probably should move this to an options class. This is the first draft of this API and given that currently we only had one option I'm guessing I did this for expediency. I'd rather defer changing it here where we're trying to merge in the changes from the already release 1.4.0-beta.1 so we have a commit in master that matches the released API.